### ISSUE TITLE
Spectating bugged when car is close to water

### ISSUE THREAD

**Issue by epbackflip** [multitheftauto/mtasa-blue#2818]
### Describe the bug

Whenever someone is spectating a player is on an object very close to water, the spectating glitches until the player gets farther from the water. The issue doesn't happen on other servers, however downgrading MTA (to version 21342) makes the issue go away.

### Steps to reproduce

Go to "Elite Players Event/Training Server" mtasa://51.38.114.131:22003 and bring someone with you
In Training Arena, search for a map which you know has objects close to water (i.e. "Loquitran", or "Greenish Flame")
Have someone spectate you or vice versa, and the issue will be present.

### Version

Client: v1.5.9-release-21415 (Windows 10. 64-bit) and some older versions
Server: MTA:SA Server v1.5.9-release-21006 (Linux 64-bit)

### Additional context

Attaching a couple of videos showing the issue:

https://streamable.com/i0f24r
https://streamable.com/w6tgbx

### Security Policy

- [X] I have read and understood the Security Policy and this issue is not security related.

**Comment by flashmta** [comment]
If you need to reproduce it on your own server:
1. Both you and the other player must be in a vehicle
2. Spectate the other player (setCameraTarget)
3. Let the other player drive close to water (preferably place an object above water)

**Comment by Lpsd** [comment]
"The issue doesn't happen on other servers"

Seems like an issue with your scripts? In which case GitHub is not suitable as this is not a core MTA issue. Please use the scripting/support channels on our community Discord. You could also use our development Discord if the talk is more related to the actual C++ code causing this.

Edit: I'd advise you use the Discord regardless of whether you believe this to be a core issue until you've actually identified this. You can also help by checking what actually changed in 21342 that may be related to this. It may not necessarily be an unintentional change in behaviour, which is why the identification needs to come first.

Edit2: [r21342](https://buildinfo.mtasa.com/?Revision=21342&Branch=) - CEF update, nothing to do with what's described here. Further confirmation that your scripts need investigation, or the actual change in behaviour is in a newer revision (you will need to confirm this).

**Comment by Lpsd** [comment]
Further to previous edits, would advise checking [r21348](https://buildinfo.mtasa.com/index.php?Revision=21348&Branch=)

**Comment by epbackflip** [comment]
Hey, reopening as I've noticed the same bug in another server (with newest MTA version however). Here's a video sample:

https://streamable.com/h10ab2

**Comment by Lpsd** [comment]
That doesn't look like the same issue to me. The only coincidence being is they're near the water, but they also collected a vehicle pickup at the same time which could be a factor.

How come the issue was closed by yourself the other day?

**Comment by epbackflip** [comment]
The bug was occurring in the same manner, when a vehicle comes close to water, no matter if a pickup is close to it or not (however this is actually the first time I noticed the bug next to a pickup).

I closed the issue earlier since I was thinking the issue was present only on my MTA Server.

**Comment by Lpsd** [comment]
Fair enough. It would really help if we can get a minimal reproduction of this issue, kind of hard to diagnose otherwise.
